polynote-frontend/polynote/ui.js

```diff
@@ -1424,48 +1442,128 @@ export class NotebookUI extends UIEventTarget {
         });
 
         socket.addMessageListener(messages.CellResult, (path, id, result) => {
+            console.log("result");
             if (path === this.path) {
                 const cell = this.cellUI.getCell(id);
-                if (cell instanceof CodeCell) {
-                    if (result instanceof CompileErrors) {
-                        cell.setErrors(result.reports);
-                    } else if (result instanceof RuntimeError) {
-                        console.log(result.error);
-                        cell.setRuntimeError(result.error);
-                    } else if (result instanceof Output) {
-                        cell.addOutput(result.contentType, result.content);
-                    } else if (result instanceof ClearResults) {
-                        cell.clearResult();
-                    } else if (result instanceof ExecutionInfo) {
-                        cell.setExecutionInfo(result);
-                    } else if (result instanceof ResultValue) {
-                        cell.addResult(result);
-                    }
+                this.handleResult(result, cell);
+            }
+        });
+    }
+
+    handleResult(result, cell) {
```

Comment:
just moved this to its own method and added a few lines for tracking results (for later building cell context for client-side interpreters)